libraries-3/src/test/java/com/baeldung/jnats/NatsClientLiveTest.java

```diff
@@ -1,115 +1,336 @@
 package com.baeldung.jnats;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
-
-import org.junit.Test;
-
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+
+import io.nats.client.Connection;
+import io.nats.client.Dispatcher;
+import io.nats.client.ErrorListener;
+import io.nats.client.JetStreamStatusException;
 import io.nats.client.Message;
-import io.nats.client.SyncSubscription;
-
+import io.nats.client.Nats;
+import io.nats.client.Options;
+import io.nats.client.Subscription;
+
+/**
+ * All the tests in this class require that a NATS server be running on localhost at the default port.
+ * See {@link <a href="https://docs.nats.io/nats-server/installation">Installing a NATS Server</a>}.
+ * <p>
+ * IMPORTANT: Awaitility.await is used to simulate the real behavior
+ * that messages take some amount of time to go from being published
+ * to being received. This amount of time will vary depending on many factors including:
+ * <ul>
+ *     <li>network latency</li>
+ *     <li>server cluster configuration</li>
+ *     <li>server computing power relative to the load of work</li>
+ * </ul>
+ * </p>
+ */
 public class NatsClientLiveTest {
 
-    @Test
-    public void givenMessageExchange_MessagesReceived() throws Exception {
+    public static final int TIMEOUT_MILLIS = 200;
+    public static final int WAIT_AT_MOST_MILLIS = 300;
+    public static final int POLL_DELAY_MILLIS = 50;
+
+    private static Connection createConnection() throws IOException, InterruptedException {
+        return createConnection("nats://localhost:4222");
+    }
 
-        NatsClient client = connectClient();
+    private static Connection createConnection(String uri) throws IOException, InterruptedException {
+        Options options = new Options.Builder().server(uri)
+            .connectionListener((connection, event) -> System.out.println("Connection Event: " + event))
```

Comment:
We use SLF4J for logging in tests, either change to a logger or remove

(similar in a few other places throughout the test)

Reply:
This is not logging. This is just to demonstrate that an event can be handled. I don't want to do logging because I don't want the user to get the idea that this is the purpose of this construct.

Reply:
The guideline we have is that, we cannot use System.out.println in an article unless the article is specifically about testing System.out

Reply:
fixed